streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -332,20 +334,31 @@ private void connectProcessorAndStateStore(String processorName, String stateSto
      *
      * @return groups of topic names
      */
-    public Map<Integer, Set<String>> topicGroups() {
-        Map<Integer, Set<String>> topicGroups = new HashMap<>();
+    public Map<Integer, PartitionGrouper.TopicsInfo> topicGroups() {
+        Map<Integer, PartitionGrouper.TopicsInfo> topicGroups = new HashMap<>();
 
         if (nodeGroups == null)
             nodeGroups = makeNodeGroups();
 
         for (Map.Entry<Integer, Set<String>> entry : nodeGroups.entrySet()) {
-            Set<String> topicGroup = new HashSet<>();
+            Set<String> sourceTopics = new HashSet<>();
+            Set<String> stateTopics = new HashSet<>();
             for (String node : entry.getValue()) {
+                // if the node is a source node, add to the source topics
                 String[] topics = nodeToTopics.get(node);
                 if (topics != null)
-                    topicGroup.addAll(Arrays.asList(topics));
+                    sourceTopics.addAll(Arrays.asList(topics));
+
+                // if the node is connected to a state, add to the state topics
+                for (Map.Entry<String, Set<String>> stateUsers : stateStoreUsers.entrySet()) {
+                    if (stateUsers.getValue().contains(node)) {
+                        stateTopics.add(stateStores.get(stateUsers.getKey()).name());
```

Comment:
I wonder how to handle a topic if a state store is used for a ktable. Should it be excluded from stateTopics?